roles/ansible-keylime-tpm20/tasks/swtpm.yml

```diff
@@ -0,0 +1,19 @@
+- name: Create swtpm directory
+  file:
+    path: /tmp/swtpm
+    state: directory
+
+- name: load swtpm module
+  command: modprobe tpm_vtpm_proxy
+
+- name: Setup swtpm
+  command: swtpm_setup --tpm2 --tpmstate /tmp/swtpm --createek --allow-signing --decryption --create-ek-cert --create-platform-cert --display
+
+- name: Run swtpm
+  command: swtpm chardev --tpm2 --vtpm-proxy --tpmstate dir=/tmp/swtpm -d
+
+- name: 'Set environment'
```

Comment:
I'm not clear on what this task is doing. It's just echo'ing out a temporary environment variable.

Reply:
It sets this as an enviromental value needed by TPM2_TOOLS. Without this the tools won't know which interface to connect to. 

https://github.com/tpm2-software/tpm2-tools/blob/master/man/common/tcti.md

Reply:
oh sorry, futher to the above its sets it for all future sessions (so you can just ssh in and not have to run export TCTI_xx everytime).

Reply:
I'm not sure it does set it for all future sessions though. The "environment" section of an ansible task is just scoped to that task. To make it permanent don't you need to write it to `/etc/environment` or global bashrc or something similar?

Reply:
let me test it..

Reply:
We may need to use something similar to [this](https://github.com/keylime/ansible-keylime-tpm-emulator/blob/df7a3745b9238c336422b9cd2dbe590b9a8267a1/quality_of_life.yml#L16) to add it to the default vagrant user's `.bashrc`. Which makes me think the keylime user will also need the same, to have the right environment.